nixos/modules/programs/steam.nix

```diff
@@ -57,6 +72,16 @@ in {
         allowedUDPPorts = [ 27015 ]; # Gameplay traffic
       })
     ];
+
+    # Append the extra compatibility packages to whatever else the env variable was populated with.
+    # For more information see https://github.com/ValveSoftware/steam-for-linux/issues/6310.
+    environment.sessionVariables = mkIf (cfg.extraCompatPackages != []) {
```

Comment:
nit: It'd be cool to eventually move this functionality into the steam package wrapper/FHSenv so that it doesn't affect the global environment, but I don't quite understand how steam is packaged/wrapped atm so I can't suggest anything.